packages/lesswrong/components/ultraFeed/AlterBonusesDialog.tsx

```diff
@@ -0,0 +1,194 @@
+import { parseNumericInputAsZeroOrNumber, customNullishCoalesceProperties, processNumericFieldInput } from './ultraFeedSettingsUtils';
+
+const styles = defineStyles('AlterBonusesDialog', (theme: ThemeType) => ({
+  dialogContent: {
+    minWidth: 500,
+    maxWidth: 700,
+    [theme.breakpoints.down('sm')]: {
+      minWidth: 'auto',
+    },
+  },
+  compactSettingGroup: {
+    backgroundColor: theme.palette.background.paper,
+    width: '100%',
+    padding: `16px 16px 0 16px`,
+    borderRadius: 4,
+  },
+  button: {
+    padding: '8px 16px',
+    fontSize: '1.1rem',
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    borderRadius: 4,
+    cursor: 'pointer',
+    border: 'none',
+    fontWeight: 500,
+  },
+  cancelButton: {
+    backgroundColor: theme.palette.grey[300],
+    color: theme.palette.text.primary,
+    '&:hover': {
+      backgroundColor: theme.palette.grey[400],
+    },
+  },
+  saveButton: {
+    backgroundColor: theme.palette.primary.main,
+    color: theme.palette.background.paper,
+    '&:hover': {
+      backgroundColor: theme.palette.primary.dark,
+    },
+  },
+  buttonDisabled: {
+    opacity: 0.5,
+    cursor: 'not-allowed',
+  },
+  actionsContent: {
+    paddingRight: 16,
+    paddingBottom: 16,
+    display: 'flex',
+    width: '100%',
+    justifyContent: 'flex-end',
+    gap: 12,
+    alignItems: 'center',
+  },
+  unsavedChangesIndicator: {
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    color: theme.palette.warning.main,
+    fontSize: 16,
+    fontStyle: 'italic',
+  },
+  buttonGroup: {
+    display: 'flex',
+    gap: 8,
+  },
+}));
+
+const AlterBonusesDialog = ({ open, onClose, className }: { open: boolean; onClose: () => void; className?: string }) => {
+  const classes = useStyles(styles);
+  const { settings, updateSettings } = useUltraFeedSettings();
+  const { flash } = useMessages();
+
+  const defaultUnifiedScoring = DEFAULT_SETTINGS.resolverSettings.unifiedScoring;
+
+  const initialFormState: UnifiedScoringFormState = useMemo(() => {
+    return mergeWith(
+      cloneDeep(defaultUnifiedScoring),
+      settings?.resolverSettings?.unifiedScoring,
+      customNullishCoalesceProperties
+    );
+  }, [settings, defaultUnifiedScoring]);
+
+  const [formValues, setFormValues] = useState<UnifiedScoringFormState>(initialFormState);
+  const [zodErrors, setZodErrors] = useState<ZodFormattedError<ValidatedUnifiedScoring, string> | null>(null);
+
+  // Reset form when dialog opens
+  useEffect(() => {
+    if (open) {
+      setFormValues(initialFormState);
+      setZodErrors(null);
+    }
+  }, [open, initialFormState]);
+
+  const handleFieldChange = useCallback((
+    field: keyof UnifiedScoringFormState,
+    value: number | string
+  ) => {
+    const processedValue = processNumericFieldInput(value);
+    setFormValues(prev => ({ ...prev, [field]: processedValue }));
+    setZodErrors(null);
+  }, []);
+
+  const handleSave = useCallback(async () => {
```

Comment:
Theoretically a bunch of this form stuff could be factored out along with the main settings stuff into some kind of hook or idk, but that's just extending this extra form system already.... I kind of expect to pull this code before too long anyhow